openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -279,12 +291,15 @@ def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_unde
                         value = flag_is_active(request, namespaced_flag_name)
                     else:
                         log.warning(u"%sFlag '%s' accessed without a request", self.log_prefix, namespaced_flag_name)
+                        set_custom_metric('warn_flag_no_request', True)
                         # Return the default value if not in a request context.
                         # Note: this skips the cache as the value might be different
                         # in a normal request context. This case seems to occur when
                         # a page redirects to a 404. In this case, we'll just return
                         # the default value.
-                        return bool(flag_undefined_default)
+                        value = bool(flag_undefined_default)
+                        self._set_waffle_flag_metric(namespaced_flag_name, value)
```

Comment:
**Reviewer Note:** This is a somewhat separate change to set the earlier-coded waffle flag metric in this special case.